app/models/ems_refresh.rb

```diff
@@ -77,6 +81,12 @@ def self.refresh(target, id = nil)
     end
   end
 
+  # Rhev specific
+  def self.refresh_new(event)
+    target = ManageIQ::Providers::Redhat::InfraManager::Vm.create_from_event(event)
```

Comment:
You don't want to be doing any provider specific operations in the base `EmsRefresh` class.

What you'll want to do is call into the provider `Refresher` class here so something like `ext_management_system.refresher.refresh_new_target` similar to how `refresh` is done [here](https://github.com/ManageIQ/manageiq/blob/master/app/models/ems_refresh.rb#L58)